### ISSUE TITLE
Design: Mockup for extension install web page

### ISSUE THREAD

**Issue by mairin** [containers/podman-desktop#2363]
Something like:
https://flathub.org/apps/io.podman_desktop.PodmanDesktop

For Podman Desktop extensions on podman-desktop.io.... list, search, install button

**Comment by ekidneyrh** [comment]
Just a small update for the layout of the extensions page. Obvs not complete but an idea of where im going with it. 

![image](https://github.com/user-attachments/assets/a2c305b5-294d-47ff-ba27-b336a0ed1326)

Have is similar to the extensions page on the app but make more emphasis on the featured extensions, much like what Flathub does on their page:

![image](https://github.com/user-attachments/assets/3fb43c0c-c7e4-4a15-82ee-65f1bcd8a827)

**Comment by ekidneyrh** [comment]
Did more work on this today:

![image](https://github.com/user-attachments/assets/56689b68-ffbd-40af-8860-c273122738eb)
[extensions web page.pdf](https://github.com/user-attachments/files/16761870/extensions.web.page.pdf)

Influenced by the flathub home page and the podman desktop extensions page. 

Need to work on the individual extensions pages next :) WDYT?

**Comment by ekidneyrh** [comment]
### Extension dashboard:

For the extensions 'dashboard' on the website, we can have a similar layout to the Flathub one linked above. We can swap out the main featured extension when we have a new one we want to highlight. In this  case it's AI Lab. There's also two smaller featured spots. I added in what we have featured in the app. 

[extensions web page.pdf](https://github.com/user-attachments/files/16782474/extensions.web.page.pdf)
![Screenshot from 2024-08-27 13-08-33](https://github.com/user-attachments/assets/c2c0395e-3f4f-4316-b6cb-0c08909eccb8)

The remaining extensions are displayed underneath in groups of three. 

### Individual extension page:

When you click any of the featured or regular extensions it will bring you to that extensions web page. Here's a sample of what the minikube one would look like:

[extension detail web page.pdf.pdf](https://github.com/user-attachments/files/16782471/extension.detail.web.page.pdf.pdf)
![extension detail web page](https://github.com/user-attachments/assets/743a2242-c12f-405b-9a08-a2ae1c940589)

I would need screen shots of screen(s) for each extension to have in the featured image area. I also just copied / cleaned up the copy from the extensions page in the app.

**Comment by benoitf** [comment]
I'll need to check if we can display the README and other resources from the webpage without hitting CORS issues

for now there is no file in the registry to feature extensions so I guess, we'll need a new one

One remark then for featured
AI Lab has a dedicated banners, how is that extension being configured ? (like is is a premium featured)

I'm asking as the webapp will iterate over JSON data, so is that we can only have "one premium" featured extension or is it the first featured being like that 

On the screen, then it's rendering nicely as we have 2 featured extensions
![image](https://github.com/user-attachments/assets/cc706a4a-836b-4cc2-a8fd-8bf2aebccbed)

what's happening if there are like 3 featured extensions to display.
We display the third one on a new line and then we go to the next line and display the other extensions

And then, how the UI is responsive.

If I have a big screen, does it still display only 3 items per row or it grows ?
and the same if I'm truing to see the page on a mobile phone

**Comment by benoitf** [comment]
at the bottom of the first image there is 1/2

does it mean that we can only display things that can be viewed on the screen ?
and then we have other pages

or do we have like infinite scrollbar (lazy load on scroll)

**Comment by ekidneyrh** [comment]
> I'll need to check if we can display the README and other resources from the webpage without hitting CORS issues
> 

No worries, If not I can go through each one and all them manually? I know it would be more effort but oh well :) 

> for now there is no file in the registry to feature extensions so I guess, we'll need a new one
> 
> One remark then for featured AI Lab has a dedicated banners, how is that extension being configured ? (like is is a premium featured)
> 

Do you mean how is it decided that it has that spot? I was thinking we just decide what extension we want to show-case. 

> I'm asking as the webapp will iterate over JSON data, so is that we can only have "one premium" featured extension or is it the first featured being like that

I think one premium and then two featured. 

> On the screen, then it's rendering nicely as we have 2 featured extensions ![image]
> 
> what's happening if there are like 3 featured extensions to display. We display the third one on a new line and then we go to the next line and display the other extensions

I would have it limited to just two on the featured line, otherwise it will be two crowded and the featured spot wont be that special. 
> And then, how the UI is responsive.

> If I have a big screen, does it still display only 3 items per row or it grows ? and the same if I'm truing to see the page on a mobile phone

I think only three that would grow when the page expands. I haven't considered mobile yet, It would probably be one per line there. The premium, featured, and normal extension card all have defining features so they will be distinguishable when stacked. I'd need to look into it more though.

**Comment by ekidneyrh** [comment]
> at the bottom of the first image there is 1/2

Thats from penpot as it's a screen shot, pls ignore :D 

> does it mean that we can only display things that can be viewed on the screen ? and then we have other pages
> 
> or do we have like infinite scrollbar (lazy load on scroll)

I dont think we have that many extensions to warrant a second page at the moment.  Flathub display 30 per page, we could do similar when our catalog gets that big.

**Comment by slemeur** [comment]
Great work @ekidneyrh ! I like the direction where it is heading.

+ 1 on the questions asked by @benoitf. 

I'd just add, that while we may not need it right away (first iteration), it would be nice to take into account the categories on the catalog. 

Then, on the detail page of the extension, I'm wondering if it could be a good idea to have an area for different links: documentation,articles/blogs posts

**Comment by vrothberg** [comment]
Drive-by comment:  I love the mock up!  Great work everybody 🥇

**Comment by ekidneyrh** [comment]
> Great work @ekidneyrh ! I like the direction where it is heading.
> 
> * 1 on the questions asked by @benoitf.
> 
> I'd just add, that while we may not need it right away (first iteration), it would be nice to take into account the categories on the catalog.

+1 to that 

> Then, on the detail page of the extension, I'm wondering if it could be a good idea to have an area for different links: documentation,articles/blogs posts

Oh that's a good idea, like links to relevant info.

I'll work on adding those in

**Comment by ekidneyrh** [comment]
Some updates!

I added in what the categories could look like - similar to Flathub and GNOME Software. I added categories such as Red Hat software and Artificial Intelligence. More can of course be added in. I would display a max of 6 extensions under each heading and then include a ‘view all’ button to look at the entirety of that category. 

![web-extensions](https://github.com/user-attachments/assets/838d2ea8-b470-4167-96f0-41a1dee4d041)

For the individual extension pages I add in a documentation and blog section in the form of cards on the right. Any docs or blogs with that extension tag will be displayed here. 

I also moved the details like version and release date to the bottom.

![web-extensions-individual](https://github.com/user-attachments/assets/9a2993da-8c5d-43a3-bc8b-e5e2ad721f0c)

**Comment by ekidneyrh** [comment]
notes from @benoitf  on Slack: 
'the information that we have for now is only https://registry.podman-desktop.io/api/extensions.json (all the data about the extensions being available to podman desktop users)
so I'm not sure that we can have for now the documentation sidebar or blog posts section
and for the top image/screenshot, we only have the README (so it's up to the extension author to add an image at the top)'

**Comment by ekidneyrh** [comment]
> notes from [@benoitf](https://github.com/benoitf) on Slack: 'the information that we have for now is only https://registry.podman-desktop.io/api/extensions.json (all the data about the extensions being available to podman desktop users) so I'm not sure that we can have for now the documentation sidebar or blog posts section and for the top image/screenshot, we only have the README (so it's up to the extension author to add an image at the top)'

I think for the docs and blog section it would display links if there is anything tagged with the ext name. In the mockup above there was a few docs with the minikube tag and no blog posts but i included it for illustrative purposes.

For the screenshot we could use a screen shot of the window when the extension open? It only has to be one so it wouldnt take that long - and the going forward we can request authors include an image to be displayed there.

**Comment by slemeur** [comment]
Should we show the version number on the landing page?

**Comment by ekidneyrh** [comment]
> Should we show the version number on the landing page?

Have them like they are in the extension catalog? 

![image](https://github.com/user-attachments/assets/8de657c0-81ad-4717-a143-49ae1c8a34de)

Without the 'already installed' or 'more details'

**Comment by slemeur** [comment]
yeah - I'm wondering if that is something really necessary though? Might be a bit too much?

**Comment by ekidneyrh** [comment]
I mean like

![image](https://github.com/user-attachments/assets/5bdccdc9-d22b-459e-8c07-d7d7bc3cad95)
